### PR TITLE
[SMALL] Fix to #5939 - Queries with "Contains" broken after upgrade from RC2 to RTM

### PR DESCRIPTION
**Pull request by maumar**
Problem was that we were not translating search expressions into non-search expressions for Extension expressions.

### CODE REVIEW

**Comment by maumar** [comment]
@smitpatel

**Comment by anpete** [comment]
:shipit:

File: src/Microsoft.EntityFrameworkCore.Relational/Query/Sql/DefaultQuerySqlGenerator.cs [hunk 1]
```diff
@@ -1745,10 +1745,26 @@ protected override Expression VisitExtension(Expression expression)
...
+
                     return expression is AliasExpression || expression is ColumnExpression || expression is SelectExpression
                         ? Expression.Equal(newExpression, Expression.Constant(true, typeof(bool)))
                         : newExpression;
                 }
+                else
+                {
+                    if (IsSearchCondition(expression))
+                    {
+                        var parentIsSearchCondition = _isSearchCondition;
```
Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/Sql/DefaultQuerySqlGenerator.cs]: Anyway to restructure logic to remove the duplication here?

Comment (maumar) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/Sql/DefaultQuerySqlGenerator.cs]: was thinking about it also, will try

**Pull request closed without merging**